### PR TITLE
[owners] JSDoc updates

### PR DESCRIPTION
**Pull request by rcebulko**
This PR:
- migrates uses of `{type[]}` to `{!Array<!type>}`
- adds `!` to `@return` declarations
- fixes some optional paramater docs

### CODE REVIEW

File: owners/src/github.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ class PullRequest {
   /**
    * Whether or not the pull request is open.
    *
-   * @return {boolean} true if the pull request is open.
+   * @return {!boolean} true if the pull request is open.
```
Comment (danielrozenberg) [inline owners/src/github.js]: I think we're not supposed to use `!` for scalar types, or for return types at all? /cc @rsimha

Comment (rcebulko) [reply owners/src/github.js]: See I thought that too re: return types but then on another PR somewhere I got called out for never using it and I couldn't find any docs saying not to so I thought I hallucinated it :thinking:

Comment (rcebulko) [reply owners/src/github.js]: As for scalars, you're right, will remove.

Comment (estherkim) [reply owners/src/github.js]: Here's a closure compiler example with `@return {!Element}`
https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System#typedefs

Comment (rsimha) [reply owners/src/github.js]: Just scalar types. See [this](https://github.com/ampproject/amphtml/blob/master/build-system/eslint-rules/closure-type-primitives.js#L147-L167) custom `amphtml` rule.

**Pull request merged**